Src/IronPython/Runtime/PythonFileManager.cs

```diff
@@ -109,17 +109,41 @@ internal sealed class StreamBox {
             return buffer;
         }
 
-        public int Write(ReadOnlySpan<byte> bytes) {
+        public int ReadInto(IPythonBuffer buffer) {
 #if NETCOREAPP
+            return _readStream.Read(buffer.AsSpan());
+#else
+            byte[]? bytes = buffer.AsUnsafeWritableArray();
+            if (bytes is not null) {
+                return _readStream.Read(bytes, 0, buffer.NumBytes());
+            }
+
+            var span = buffer.AsSpan();
```

Comment:
If we're moving this code into a helper that might end up being called elsewhere it's probably worth optimizing it a bit.

Reply:
Thanks, works for me! Another option would have been to go with the base `Stream.Read(Span<byte>)` implementation:
```cs
        private static int Read(this Stream stream, Span<byte> buffer) {
            byte[] sharedBuffer = ArrayPool<byte>.Shared.Rent(buffer.Length);
            try {
                int numRead = stream.Read(sharedBuffer, 0, buffer.Length);
                if ((uint)numRead > (uint)buffer.Length) {
                    throw new IOException();
                }

                new ReadOnlySpan<byte>(sharedBuffer, 0, numRead).CopyTo(buffer);
                return numRead;
            } finally {
                ArrayPool<byte>.Shared.Return(sharedBuffer);
            }
        }
```